addons/stock/models/stock_quant.py

```diff
@@ -280,12 +280,22 @@ def create(self, vals_list):
     def _load_records_create(self, values):
         """ Add default location if import file did not fill it"""
         company_user = self.env.company
+        self = self.with_context(inventory_mode=True)
         warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_user.id)], limit=1)
         for value in values:
             if 'location_id' not in value:
                 value['location_id'] = warehouse.lot_stock_id.id
         return super()._load_records_create(values)
 
+    def _load_records_write(self, values):
+        """ Only allowed fields should be modified """
+        self = self.with_context(inventory_mode=True)
```

Comment:
Now that you've added this line, won't the `def write` logic handle the check for the not allowed_fields? It's not a big deal if we double check the condition, it just might create redundant code.

Reply:
Also since we closed the original PR's forward ports, you will need to add update the .pot files in the new forward ports if you do end up adding the new UserError

Reply:
@ticodoo I'll re-check ``_load_records_write`` and see, for the `.pot` file I'd prefer to get the chain merged and then make a new PR form saas-15.x as previous PRs are missing it too

Reply:
@ticodoo ironically in `saas-15.3` the issue is solved so this PR is pointless unless we want to make the error message clear to match what's in `15.0` also if you still want to have `self.with_context(inventory_mode=True)` in create method. Let me know what you think.

Reply:
Weird, any idea why it's not broken in saas-15.3? I would rather not change it if we don't need to :thinking:

Reply:
@ticodoo I couldn't find the commit that fixed the issue but meh I'll have to revert `saas-15.2` as the issue is not triggered w/o the previous PR